app/models/user.rb

```diff
@@ -1,6 +1,24 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  paginates_per 5
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+  has_many :friendships
+  has_many :friends, through: :friendships
+
+  def self.all_except(user)
+    where.not(id: Friendship.where(user_id: user.id).pluck(:friend_id) << user.id)
+  end
+  
+  def self.friends(user)
+    where(id: Friendship.where(user_id: user.id, status: true).pluck(:friend_id))
```

Comment:
Since we have both directional records in `friendships` table. We can do simple `User.first.friends`, it should automatically return friends of a user, no need to make these two queries.

Reply:
User.first.friends gets all friends to home I have sent friend request.
for User.first.friends => where user_id: User.first.id, status: false/ true
what i need for friends => where user_id: User.first.id, status: true